### PR TITLE
Pack should support run image extension on Platform API 0.12

### LINKED ISSUE #1620: [RFC #0105] pack should support run image extension
When there is a lifecycle that supports run image extension, pack should also support run image extension by:

* Invoking the `extender` with `-kind run` when there are Dockerfiles in `<generated>/run` (output of the `detect` phase)

See related lifecycle issue: https://github.com/buildpacks/lifecycle/issues/998

### PR DESCRIPTION
**Pull request by natalieparellano**
Fixes #1620

### CODE REVIEW

File: internal/build/lifecycle_execution.go [hunk 1]
```diff
@@ -35,6 +38,7 @@ type LifecycleExecution struct {
 	os           string
 	mountPaths   mountPaths
 	opts         LifecycleOptions
+	WorkingDir   string
```
Comment (natalieparellano) [inline internal/build/lifecycle_execution.go:41]: I don't love making this public just for tests. But the alternatives were a bit more complicated. If anyone has feedback here, I'd be glad to hear it.

File: internal/build/lifecycle_execution.go [hunk 2]
```diff
@@ -35,6 +38,7 @@ type LifecycleExecution struct {
 	os           string
 	mountPaths   mountPaths
 	opts         LifecycleOptions
+	TmpDir       string
```
Comment (natalieparellano) [inline internal/build/lifecycle_execution.go:41]: I don't love making this public just for tests. But the alternatives were a bit more complicated. If anyone has feedback here, I'd be glad to hear it.

File: acceptance/acceptance_test.go [hunk 3]
```diff
@@ -798,7 +798,7 @@ func testAcceptance(
 					it("creates builder", func() {
 						// Linux containers (including Linux containers on Windows)
 						extSimpleLayersDiffID := "sha256:b9e4a0ddfb650c7aa71d1e6aceea1665365e409b3078bfdc1e51c2b07ab2b423"
-						extReadEnvDiffID := "sha256:b49f178f802bbeb04acf4776bd41dcea8c47504ce31b06b580ec697387629cf2"
+						extReadEnvDiffID := "sha256:ab7419c5e0b1a0789bd07cef2ed0573ec6e98eb05d7f05eb95d4f035243e331c"
```
Comment (natalieparellano) [inline acceptance/acceptance_test.go:801]: Changed the fixture, so the SHA changed

File: acceptance/acceptance_test.go [hunk 4]
```diff
@@ -853,7 +853,37 @@ func testAcceptance(
...
+								inspectCmd := "inspect"
+								if !pack.Supports("inspect") {
+									inspectCmd = "inspect-image"
+								}
+
+								output = pack.RunSuccessfully(inspectCmd, repoName)
+							})
+						})
+
+						when("there are run image extensions", func() {
```
Comment (natalieparellano) [inline acceptance/acceptance_test.go:868]: This test won't run in CI until there's a lifecycle that supports run image extension. But I've verified with my dev lifecycle that it does the job.

File: internal/build/lifecycle_execution.go [hunk 5]
```diff
@@ -655,7 +718,7 @@ func (l *LifecycleExecution) Export(ctx context.Context, buildCache, launchCache
 	flags := []string{
 		"-app", l.mountPaths.appDir(),
 		"-cache-dir", l.mountPaths.cacheDir(),
-		"-stack", l.mountPaths.stackPath(),
+		// "-stack", l.mountPaths.stackPath(), // TODO: remove, replace with "-run" as part of https://github.com/buildpacks/pack/issues/1301
```
Comment (joe-kimmel-vmw) [inline internal/build/lifecycle_execution.go:721]: probably  you meant to do some resolution here

File: internal/build/lifecycle_execution.go (hunk 2 above)
Comment (jkutner) [reply internal/build/lifecycle_execution.go:41]: We could have a `NewLifecycleExecutionWithTmpDir` that accepts it as an arg.

File: internal/build/container_ops.go [hunk 6]
```diff
@@ -42,6 +43,30 @@ func CopyOut(handler func(closer io.ReadCloser) error, srcs ...string) Container
 	}
 }
 
+// CopyOutErr copies container directories to a handler function. The handler is responsible for closing the Reader.
+// CopyOutErr differs from CopyOut in that it does not require the source files to exist in the container
+// in order to exit without error.
+func CopyOutErr(handler func(closer io.ReadCloser) error, srcs ...string) ContainerOperation {
```
[the remark this answers is not in the document]
Comment (jkutner) [reply internal/build/container_ops.go:49]: I added a test to this branch/PR d6aedda62a44c90b5e3c4b6e8ad46f0b8bc93d16 to try and cover this. But I didn't understand the statement well enough to write a test "does not require the source files to exist in the container in order to exit without error".

I think _maybe_ `CopyOutMaybe` is a better name? I'd like a test for this too, but I'm struggling to understand it.

File: internal/build/lifecycle_execution.go (hunk 5 above)
Comment (natalieparellano) [reply internal/build/lifecycle_execution.go]: Updated in e8c869a7

File: internal/build/lifecycle_execution.go (hunk 2 above)
Comment (natalieparellano) [reply internal/build/lifecycle_execution.go]: Sounds good - I just updated (in 15e52db0) the existing constructor to have a new argument (it's only called once in an internal package).

**Review by jkutner — APPROVED** [review]

**Pull request merged**